src/Swarm/Language/Typecheck.hs

```diff
@@ -384,6 +384,7 @@ infer s@(Syntax l t) = (`catchError` addLocToTypeErr s) $ case t of
     ty2 <- maybe id (`withBinding` Forall [] a) mx $ infer c2
     _ <- decomposeCmdTy ty2
     return ty2
+  TNonCode {} -> return UTyInt -- TODO: probably wrong, but I do not understand this code
```

Comment:
There is nothing sensible to do here because there is no way to know what the type of a `TNonCode` is.  Another reason that we should get rid of it. =)